### PR TITLE
Guard and cache fetchPermission chrome API.

### PR DESCRIPTION
**Pull request by Hyperkid123**
jira: https://issues.redhat.com/browse/RHCLOUD-9200

### Changes
- adds a guard for `fetchPermission` chrome API call
  - invoking fetchPermission x > 1 times will always result only in one API call.
- use maximum page size for the `fetchPermission` config
  - previously the page size was 25, now it's 100
  - this change alone reduces the number of calls 4x
- cache the `fetchPermission` call
  - this will remove many rbac API calls when switching between apps
  - the cache is invalidated after 10 minutes

### CODE REVIEW

File: src/js/rbac/fetchPermissions.js [hunk 1]
```diff
@@ -23,3 +22,28 @@ export const fetchPermissions = (userToken, app = '') => {
         }})
     .catch(error => log(error));
 };
+
+export const createFetchPermissionsWatcher = (getUser, libjwt) => {
+    let currentCall = undefined;
+    const cache = new CacheAdapter(
```
Comment (karelhala) [inline src/js/rbac/fetchPermissions.js:28]: I wouldn't create new cache, I'd rather share chrome cache from all over places, meaning passing it to `createFetchPermissionsWatcher`
```JS
const createFetchPermissionsWatcher = (cache) => {
  let currentCall = undefined;
    return async (userToken, app = '') => {
        if (insights.chrome.getBundle() === 'openshift') {
            return Promise.resolve([]);
        }
        const permissions = await cache.getItem('permissions');
        if (permissions) {
            return permissions;
        }
        if (typeof currentCall === 'undefined') {
            currentCall = getUser().then(() => fetchPermissions(userToken, app).then((data) => {
                currentCall = undefined;
                cache.setItem('permissions', data);
                return data;
            }));
        }
        return currentCall;
    };
}
```

File: src/js/entry.js [hunk 2]
```diff
@@ -128,27 +128,29 @@ export function chromeInit(libjwt) {
...
+
+        qe.init();
+
+        return libjwt.initPromise
+        .then(libjwt.jwt.getUserInfo)
+        .catch(() => {
+            libjwt.jwt.logoutAllTabs();
+        });
+    };
+    const fetchPermissions = createFetchPermissionsWatcher(getUser, libjwt);
```
Comment (karelhala) [inline src/js/entry.js:146]: I'd change this to consume only cahce
```JS
const fetchPermissions = getUser().then(() => createFetchPermissionsWatcher(cache));
```

Comment (karelhala) [reply src/js/entry.js:146]: In order to make it work correctly you'd have to moce `let chromeCache` on line 56 move above chrome bootstrap.

File: src/js/rbac/fetchPermissions.js (hunk 1 above)
Comment (Hyperkid123) [reply src/js/rbac/fetchPermissions.js:28]: @kate that would be perfect however that will take a little bit more effort than just lifting to a new file which will delay this fix. I would like to do that in a different PR.

**Review by karelhala — DISMISSED** [review]

**Review by karelhala — APPROVED** [review]

**Comment by Hyperkid123** [comment]
@karelhala I have forgotten to update the tests.

**Comment by Hyperkid123** [comment]
@karelhala also using the global cache does not work :cry: The endpoint is called before the cache is created... Let's use the local cache for now and I will come up with a solution to make the chrome cache global in a different PR.
This is the error. The cache is, unfortunately `undefined`.
```
fetchPermissions.js:69 Uncaught (in promise) TypeError: Cannot read property 'getItem' of undefined
    at _callee$ (fetchPermissions.js:69)
    at tryCatch (runtime.js:63)
    at Generator.invoke [as _invoke] (runtime.js:293)
    at Generator.eval [as next] (runtime.js:118)
    at asyncGeneratorStep (asyncToGenerator.js:3)
    at _next (asyncToGenerator.js:25)
    at eval (asyncToGenerator.js:32)
    at new Promise (<anonymous>)
    at eval (asyncToGenerator.js:21)
    at eval (fetchPermissions.js:101)
```

**Review by karelhala — APPROVED** [review]

**Pull request merged**